be/src/formats/csv/string_converter.cpp

```diff
@@ -55,30 +59,33 @@ Status StringConverter::write_quoted_string(OutputStream* os, const Column& colu
 }
 
 bool StringConverter::read_string(Column* column, const Slice& s, const Options& options) const {
-    int max_size = 0;
+    size_t max_size = 0;
     if (options.type_desc != nullptr) {
         max_size = options.type_desc->len;
     }
 
-    bool length_check_status = true;
-    // Hive table, not limit string length <= 1mb anymore
     if (options.is_hive) {
-        if (UNLIKELY(max_size > 0 && s.size > max_size)) {
-            length_check_status = false;
+        // truncate directly, support for utf-8 encoding
+        std::vector<size_t> index{};
+        const size_t utf8_length = get_utf8_index(s, &index);
+        size_t actual_size = 0;
+        if (utf8_length > max_size) {
+            // do truncate
+            actual_size = index[max_size];
```

Comment:
after truncation, is it a still valid utf8?

Reply:
yes